README.md

```diff
@@ -201,6 +201,219 @@ hints and first-parties can use `Feature-Policy` to decide which third-parties t
+
+### OS integration
+Similarly, some sites would want to change links to OS specific ones (e.g.
+[Android intent
+links](https://developer.chrome.com/multidevice/android/intents)).  While,
+again, progressive enhancement can be used to modify those links using script,
+rather than bake them into the HTML, some sites may prefer server-side
+adaptation.
+
+Again, like the "OS specific styles" case, they'd need to use the platform and
+platform version hints to do so.
+
+### Browser and OS specific experiments
+Some servers may like to limit their multi variant experimentation to specific
+browsers, specific platforms or specific versions of any of the above.  For
+experiments that are limited to browser and version, those sites can use the
+`Sec-CH-UA` values sent by default on requests.  If they require platform and
+its version, they'd have to opt-in for those hints, or use client-side scripts
+to control the experimentation.
+
+## User login notification
+Many sites, especially security sensitive ones, like to notify their users when
+log-in from a new device happens. That enables users to be aware of those
+logins, and take action in case it's not a login that's done by them or on
+their behalf.
+
+For those notifications to be meaningful, sites need to recognize and
+communicate the commercial brand of the browser to the user. These messages
+often also include the platform and its version in order to make sure the user
+knows which device is in question.
+
+Since such messaging doesn't require any server-side adaptation, it's better
+for this case to use the `getUserAgent` method in order to retrieve the
+  required information.
+
+## Conversion modeling
+Some machine learning models use various details from the `User-Agent` string
+in order to estimate various things about users of those user agents.  Similar
+modeling would still be possible, but will require explicit opt-in to collect
+the required bits of information.
+
+## Vulnerability filtering
+In some environments, proxy servers may be used to verify that the different
+users accessing information are not doing so from obsolete devices, that are
+potentialy vulnerable to security issues.  While the browser and version
+information available from `Sec-CH-UA` can provide some information, the
+browser and OS full version are often useful for that kind of analysis.
+
+Such proxies would have to add a redirect step that opts-in to getting the
+browser full version and the platform version in order to continue to get
+access to those hints.
+
+## Logs and debugging
+Many services log the `User-Agent` string today and can use it in various ways
+when analyzing past traffic or when trying to debug errors related to their
+service.  Those services will have to use the lower entropy values available
+through `Sec-CH-UA` for logging purposes, or opt-in to receive higher-entropy
+hints. The latter doesn't seem like something services should do just for
+forensic purposes.
```

Comment:
Although when a user selects "Send Feedback", it might make sense to use the JS functions to collect more information at that point.

Reply:
Good point. Added a comment to that effect.